stdlib/array.mli

```diff
@@ -320,6 +320,71 @@ val of_seq : 'a Seq.t -> 'a array
 (** Create an array from the generator
     @since 4.07 *)
 
+(** {1:array_concurrency Arrays and concurrency safety}
+
+    Care must be taken when concurrently accessing arrays from multiple domains.
+
+    {2:array_atomicity Atomicity}
+
+    Any array operation that accesses more than one element is not guaranteed to
+    be atomic in the presence of concurrent accesses. This includes iteration,
```

Comment:
Since all functions in the `Array` module are not atomic, I would simplify the sentence to:
```
All functions in the `Array` module that access more than one element are not atomic in the presence of unsynchronized accesses.
```

Reply:
I believe that "all ... are not" is not idiomatic. How about "Every array operation that accesses more than one array element is *not* atomic...".

Or, alternatively, give an explicit list of the operations that *are* atomic, and indicate that every function that does not appear in this list is not atomic.

Reply:
> Since all functions in the Array module are not atomic, I would simplify the sentence to:

I think we have to be careful that there are two notions of atomicity here.

1. Atomic memory locations in the memory model sense. 
2. Atomic operations which combine multiple non-atomic operations together (using say `Mutex.t`). 

The phrase "All functions in the Array module are not atomic" is true, but `Array.get` is not atomic in the sense of (1) whereas `Array.iter` is not atomic in the sense of (2). Here, I am talking about (2). 

If we just want to rephrase the sentence, then we should go with @fpottier's suggestion

> Every array operation that accesses more than one array element is not atomic.

Reply:
That sounds good to me.